src/nanoarrow/ipc/encoder.c

```diff
@@ -78,30 +78,33 @@ void ArrowIpcEncoderReset(struct ArrowIpcEncoder* encoder) {
   memset(encoder, 0, sizeof(struct ArrowIpcEncoder));
 }
 
+static ArrowErrorCode ArrowIpcEncoderWriteContinuationAndSize(struct ArrowBuffer* out,
+                                                              size_t size) {
+  _NANOARROW_CHECK_UPPER_LIMIT(size, INT32_MAX);
+  int32_t header[] = {-1, ArrowIpcSystemEndianness() == NANOARROW_IPC_ENDIANNESS_BIG
+                              ? (int32_t)bswap32((uint32_t)size)
+                              : (int32_t)size};
+  return ArrowBufferAppend(out, &header, sizeof(header));
```

Comment:
`header` works because it's an array and arrays are convertible to pointer. I prefer `&header` because that would be correct even if `header` were not an array